### PR TITLE
Add support (and a requirement) for aeson-2.

### PR DESCRIPTION
**Pull request by athas**
This is only interesting because krank is currently broken in Nixpkgs,
and this will fix it.  That is also why the PR contains an update of
nixpkgs.nix.

### CODE REVIEW

**Review by clkamp — APPROVED** [review]
LGTM

**Comment by guibou** [comment]
LGTM, however I'm wondering if it will work with older versions of GHC / `base` / ...

Let's see what the unit tests are saying.

**Comment by athas** [comment]
The Nix builds fail for CI reasons.  I don't understand why Stack fails; I thought I fixed that.  I'll take a look.

**Comment by athas** [comment]
Apparently CI does not use stack.yaml.  Is it important to maintain compatibility with GHC 8.8?  Isn't krank mostly an executable and not a library?

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -32,6 +32,4 @@ jobs:
 
     # Build stack
     - name: Stack 8.10
-      run: stack --resolver lts-17.1 test
-    - name: Stack 8.8
-      run: stack --resolver lts-16.31 test
+      run: stack --resolver lts-19.6 test
```
Comment (guibou) [inline .github/workflows/main.yaml:35]: Could you also rename the test name above, to GHC 9.0?

**Comment by guibou** [comment]
> Apparently CI does not use stack.yaml. Is it important to maintain compatibility with GHC 8.8? Isn't krank mostly an executable and not a library?

Actually, you are right, I don't really care about compatibility with older GHC as long as:

- It works with whatever the default is with stack
- It works with the default GHC of main distros

File: .github/workflows/main.yaml (hunk 1 above)
Comment (athas) [reply .github/workflows/main.yaml:35]: Done.

**Comment by athas** [comment]
This PR doesn't really change GHC compatibility (aeson 2 works with very old GHCs), and the current stack default is aeson 2.

**Comment by guibou** [comment]
Thank you!

**Comment by athas** [comment]
Great! As soon as a new version is also on Hackage, we can activate it again in Nixpkgs.

**Pull request merged**